### ISSUE TITLE
Feature request: support matrix RHS for cupyx.scipy.sparse.linalg.spsolve

### ISSUE THREAD

**Issue by grlee77** [cupy/cupy#4665]
`scipy.sparse.linalg.spsolve` supports the case where the right-hand side (RHS), `b` is a matrix. Can we also support that case in CuPy?

It looks like this function uses the [cuSOLVER csrlsvqr methods](https://docs.nvidia.com/cuda/cusolver/index.html#cusolver-lt-t-gt-csrlsvqr) which only look like they support vector `b`. In that case, should we just add a loop over the other axis if a matrix `b` is provided?

There are some `csrqrBatched` methods in `cuSOLVER`, but those look like they are designed for the case where one also has a batch of CSR matrices (i.e. one CSR matrix for each vector in the RHS rather than the same matrix for all of them).

**Comment by leofang** [comment]
> There are some `csrqrBatched` methods in `cuSOLVER`, but those look like they are designed for the case where one also has a batch of CSR matrices (i.e. one CSR matrix for each vector in the RHS rather than the same matrix for all of them).

Yeah looks like the API of [`cusolverSpXcsrqrBatched()`](https://docs.nvidia.com/cuda/cusolver/index.html#cusolver-lt-t-gt-csrqrbatched) has a nonideal design. In a few batched solvers like `potrf`/`potrs`, the input is a pointer to pointer instead of a plain pointer, so in that case the flexibility is enough to meet our need here, as we could just repeat the pointer to `csrValA` for `batchSize` times without any copy...I think a simple C loop as being done in #4628 for batched SVD is a reasonable start.
